cli/lib/metadata.ts

```diff
@@ -117,11 +138,15 @@ export async function injectMetadata(this: Context, jobUri: string) {
 
     callback()
   }, function (callback) {
-    for (const [propertyShape, dimension] of propertyShapes) {
-      if (versionedDimensions.has(dimension)) {
-        this.push($rdf.quad(propertyShape, schema.version, revision))
+    publishTracer.startActiveSpan('injectMetadata#finished', span => {
```

Comment:
it usually does not make sense to create span for synchronous code like this, since it will only last a very short amount of time.
Maybe you want to have a bigger span that wraps all this and log additional events with [`span.addEvent()`](https://open-telemetry.github.io/opentelemetry-js-api/interfaces/span.html#addevent)? Also note that you can dynamically add attributes to spans with [`span.setAttribute()`](https://open-telemetry.github.io/opentelemetry-js-api/interfaces/span.html#setattribute)